owncloudApp/src/main/java/com/owncloud/android/presentation/files/removefile/RemoveFilesDialogFragment.kt

```diff
@@ -22,110 +22,127 @@
                     R.string.confirmation_remove_files_alert
                 }
             }
-            val localRemoveButton = if (isAvailableLocallyAndNotAvailableOffline) {
-                R.string.confirmation_remove_local
+            dialogRemoveInformation.text = requireContext().getString(messageStringId, messageArguments)
+
+            if (isAvailableLocallyAndNotAvailableOffline) {
+                dialogRemoveLocalOnly.text = requireContext().getString(R.string.confirmation_remove_local)
             } else {
-                -1
+                dialogRemoveLocalOnly.visibility = View.GONE
             }
 
-            val args = Bundle().apply {
-                putInt(ARG_MESSAGE_RESOURCE_ID, messageStringId)
-                if (files.size == 1) {
-                    putStringArray(ARG_MESSAGE_ARGUMENTS, arrayOf(files.first().fileName))
-                } else {
-                    putStringArray(ARG_MESSAGE_ARGUMENTS, arrayOf(files.size.toString()))
-                }
-                putInt(ARG_POSITIVE_BTN_RES, R.string.common_yes)
-                putInt(ARG_NEUTRAL_BTN_RES, R.string.common_no)
-                putInt(ARG_NEGATIVE_BTN_RES, localRemoveButton)
-                putParcelableArrayList(ARG_TARGET_FILES, files)
+            dialogRemoveLocalOnly.setOnClickListener {
+                fileOperationViewModel.performOperation(FileOperation.RemoveOperation(targetFiles.toList(), removeOnlyLocalCopy = true))
+                dismiss()
             }
 
-            return RemoveFilesDialogFragment().apply {
-                arguments = args
+            dialogRemoveYes.setOnClickListener {
+                fileOperationViewModel.performOperation(FileOperation.RemoveOperation(targetFiles.toList(), removeOnlyLocalCopy = false))
+                dismiss()
             }
+
+            dialogRemoveNo.setOnClickListener { dismiss() }
+        }
+    }
+
+    private fun handleThumbnail(files: List<OCFile>, thumbnailImageView: ImageView) {
+        if (files.size == 1) {
+            val file = files[0]
+            // Show the thumbnail when the file has one
+            val thumbnail = ThumbnailsCacheManager.getBitmapFromDiskCache(file.remoteId)
+            if (thumbnail != null) {
+                thumbnailImageView.setImageBitmap(thumbnail)
+            } else {
+                thumbnailImageView.setImageResource(
+                    MimetypeIconUtil.getFileTypeIconId(file.mimeType, file.fileName)
+                )
+            }
+        } else {
+            thumbnailImageView.visibility = View.GONE
+        }
+    }
+
+    companion object {
+        const val TAG_REMOVE_FILES_DIALOG_FRAGMENT = "TAG_REMOVE_FILES_DIALOG_FRAGMENT"
+
+        /**
+         * Public factory method to create new RemoveFilesDialogFragment instances.
+         *
+         * @param files           Files to remove.
+         * @param isAvailableLocallyAndNotAvailableOffline
+         * @return Dialog ready to show.
+         */
```

Comment:
Same as in other PRs: if you want to keep the docs, do it correctly. `isAvailableLocallyAndNotAvailableOffline` is not described